classes/scanner.php

```diff
@@ -275,19 +277,37 @@ protected function is_libreoffice_encrypted(string $file) : bool {
     }
 
     protected function is_pdf_encrypted(string $file) : bool {
+        global $CFG;
+
+        // Run file through ghostscript to ensure no encrpytion.
+        $new = $file . '_clean';
+        $gsexec = \escapeshellarg($CFG->pathtogs);
+        $path = \escapeshellarg($file);
+        $newpath = \escapeshellarg($new);
+        $command = "$gsexec -q -sDEVICE=pdfwrite -dBATCH -dNOPAUSE -sOutputFile=$newpath $path";
```

Comment:
1) just thinking of ways to make this as fast as possible and reduce IO. ```sOutputFile-``` should pipe it straight to dev null without the second file.

2) If someone uploads a massive pdf which is fine, printing just the first page should be enough of a test, maybe this ```-dFirstPage=1 -dLastPage=1``` (not tested, random googling)

3) Also can we rely on this completely and remove the editpdf\pdf which in theory is redundant now?